### DISCUSSION TITLE
Add perturbations to PSI simulation

### DISCUSSION THREAD

**Discussion by jd-lara** [Sienna-Platform/PowerSimulations.jl#870] [Ideas]
To perform consistent reliability analysis with PSI, we need to implement a perturbation concept that performs the same feature as running a callback in PSID.

**Comment by yasirroni** [comment]
One of the point from the discussions (request) with @kdayday is to support cascading trips from the defined perturbation, including early stopper when the simulation got no solution at time `t` due to the cascading.

A user defined perturbation (that is not cascading) is supported at: https://sienna-platform.github.io/Sienna/SiennaDocs/docs/build/PowerSimulationsDynamics/stable/perturbations/

**Reply by jd-lara** [reply]
we probably need an issue with a more detailed description to cover this in PSI
